aligning `mongo_uri/database` breaks connecting to Atlas cluster

[This change](https://github.com/Materials-Consortia/optimade-python-tools/commit/03cd11c065eb73beed4ee240af7f65eca3648866) in handling `MONGO_URI` and `MONGO_DATABASE` breaks connecting to a MongoDB Atlas Cluster via its connection string.

For instance, setting the env vars

```
# database set explicitly, no database name in Mongo URI
OPTIMADE_MONGO_DATABASE=<database-name>
OPTIMADE_MONGO_URI=mongodb+srv://<username>:<password>@<cluster-name>.XXXXX.mongodb.net

# OR
# only set Mongo URI with database name included
OPTIMADE_MONGO_URI=mongodb+srv://<username>:<password>@<cluster-name>.XXXXX.mongodb.net/<database-name>
```

both result in this error:

```
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/optimade/server/config.py", line 501, in align_mongo_uri_and_mongo_database
optimade-1  |     uri: dict[str, Any] = pymongo.uri_parser.parse_uri(self.mongo_uri)
optimade-1  |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/pymongo/uri_parser.py", line 524, in parse_uri
optimade-1  |     raise InvalidURI('Bad database name "%s"' % dbase)
optimade-1  | pymongo.errors.InvalidURI: Bad database name "/<username>:<password>@<cluster-name>"
```

Since `ServerConfig.mongo_database` [defaults](https://github.com/Materials-Consortia/optimade-python-tools/blob/9af05bcec38de3318fd28bae1d8a8f7d15e18c80/optimade/server/config.py#L229) to `optimade`, `self.mongo_uri and self.mongo_database` [here](https://github.com/Materials-Consortia/optimade-python-tools/blob/9af05bcec38de3318fd28bae1d8a8f7d15e18c80/optimade/server/config.py#L493) is `True` even when `OPTIMADE_MONGO_DATABASE` is not set in the environment. Explicitly setting `OPTIMADE_MONGO_DATABASE` to an empty string raises the standard pymongo error:

```
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/pymongo/database_shared.py", line 27, in _check_name
optimade-1  |     raise InvalidName("database name cannot be the empty string")
optimade-1  | pymongo.errors.InvalidName: database name cannot be the empty string
```

HTH and hope I didn't miss anything obvious :)

The issue remains the same with v1.1.7. The following traceback is for setting `OPTIMADE_MONGO_URI` but omitting `OPTIMADE_MONGO_DATABASE` (it'll default to `optimade`). Atlas has a free tier which should suffice to test this.

```
optimade-1  | OPTIMADE_MONGO_URI=mongodb+srv://<username>:<password>@<host>/<db>
optimade-1  | OPTIMADE_MONGO_DATABASE=optimade
optimade-1  | OPTIMADE_CONFIG_FILE=/app/config.json
optimade-1  | [2024-12-16 20:19:14 +0000] [7] [INFO] Starting gunicorn 23.0.0
optimade-1  | [2024-12-16 20:19:14 +0000] [7] [INFO] Listening at: http://0.0.0.0:10004 (7)
optimade-1  | [2024-12-16 20:19:14 +0000] [7] [INFO] Using worker: uvicorn.workers.UvicornWorker
optimade-1  | [2024-12-16 20:19:14 +0000] [10] [INFO] Booting worker with pid: 10
optimade-1  | [2024-12-16 20:19:15 +0000] [10] [ERROR] Exception in worker process
optimade-1  | Traceback (most recent call last):
[...]
optimade-1  |   File "/app/app.py", line 5, in <module>
optimade-1  |     from optimade.server.main import app
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/optimade/server/main.py", line 18, in <module>
optimade-1  |     from optimade.server.config import CONFIG, DEFAULT_CONFIG_FILE_PATH
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/optimade/server/config.py", line 543, in <module>
optimade-1  |     CONFIG: ServerConfig = ServerConfig()
optimade-1  |                            ~~~~~~~~~~~~^^
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/pydantic_settings/main.py", line 171, in __init__
[...]
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/pydantic/main.py", line 214, in __init__
optimade-1  |     validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/optimade/server/config.py", line 501, in align_mongo_uri_and_mongo_database
optimade-1  |     uri: dict[str, Any] = pymongo.uri_parser.parse_uri(
optimade-1  |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
optimade-1  |         self.mongo_uri, warn=True
optimade-1  |         ^^^^^^^^^^^^^^^^^^^^^^^^^
optimade-1  |     )
optimade-1  |     ^
optimade-1  |   File "/root/.local/lib/python3.13/site-packages/pymongo/uri_parser.py", line 524, in parse_uri
optimade-1  |     raise InvalidURI('Bad database name "%s"' % dbase)
optimade-1  | pymongo.errors.InvalidURI: Bad database name "/<username>:<password>@<host>"
```

--- a/optimade/server/config.py
+++ b/optimade/server/config.py
@@ -498,16 +498,11 @@ class ServerConfig(BaseSettings):
                     or self.mongo_uri.startswith("mongodb+srv://")
                 ):
                     self.mongo_uri = f"mongodb://{self.mongo_uri}"
-                try:
-                    uri: dict[str, Any] = pymongo.uri_parser.parse_uri(
-                        self.mongo_uri, warn=True
-                    )
-                    if uri.get("database"):
-                        self.mongo_database = uri["database"]
-                except pymongo.errors.InvalidURI as error_msg:
-                    warnings.warn(
-                        f"The uri {self.mongo_uri} may be invalid.\n{error_msg}"
-                    )
+                uri: dict[str, Any] = pymongo.uri_parser.parse_uri(
+                    self.mongo_uri, warn=True
+                )
+                if uri.get("database"):
+                    self.mongo_database = uri["database"]
 
         return self
 

--- a/optimade/server/config.py
+++ b/optimade/server/config.py
@@ -493,9 +493,10 @@ class ServerConfig(BaseSettings):
             if self.mongo_uri:
                 import pymongo.uri_parser
 
-                if not self.mongo_uri.startswith(
-                    "mongodb://"
-                ) or self.mongo_uri.startswith("mongodb+srv://"):
+                if not (
+                    self.mongo_uri.startswith("mongodb://")
+                    or self.mongo_uri.startswith("mongodb+srv://")
+                ):
                     self.mongo_uri = f"mongodb://{self.mongo_uri}"
                 try:
                     uri: dict[str, Any] = pymongo.uri_parser.parse_uri(

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -167,4 +167,5 @@ filterwarnings = [
     "ignore:.*The 'app' shortcut is now deprecated.*:DeprecationWarning", # Raised by httpx for test client code; would rather not force upgrade of httpx to deal with it for now
     "ignore::pytest.PytestUnraisableExceptionWarning",  # Started raising with pytest 8.2.1, seemingly related to anyio warnings
     "ignore:.*Unclosed.*:ResourceWarning", # Also started raising with pytest 8.2.1, seemingly related to anyio warnings
+    "ignore:.*This function should have been removed on 2025-01-01*:DeprecationWarning", # Raised by monty, who knows when it will actually be removed
 ]

--- a/tests/adapters/conftest.py
+++ b/tests/adapters/conftest.py
@@ -5,6 +5,13 @@ import pytest
 import requests
 
 
+@pytest.fixture(autouse=True)
+def set_ci_env(monkeypatch):
+    """Super unpleasant workaround to deal with monty's deprecation behaviour."""
+    monkeypatch.setenv("CI", "false")
+    yield
+
+
 @pytest.fixture
 def mock_requests_get(monkeypatch):
     """Patch requests.get to return the desired mock response."""

